exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Alias.kt

```diff
@@ -54,9 +59,54 @@ class Alias<out T : Table>(val delegate: T, val alias: String) : Table() {
     override fun hashCode(): Int = tableNameWithAlias.hashCode()
 
     @Suppress("UNCHECKED_CAST")
-    operator fun <T : Any?> get(original: Column<T>): Column<T> =
-        delegate.columns.find { it == original }?.let { it.clone() as? Column<T> }
+    operator fun <T : Any?> get(original: Column<T>): Column<T> {
+        // CompositeIdTable id is not a typical database-registered column
+        val delegateColumn = if (delegate is CompositeIdTable && original.columnType.isEntityIdentifier()) {
+            delegate.id
+        } else {
+            delegate.columns.find { it == original }
+        }
+        return delegateColumn?.let { it.clone() as? Column<T> }
             ?: error("Column not found in original table")
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    override fun mapIdComparison(
+        toCompare: Any?,
+        operator: (Column<*>, Expression<*>) -> Op<Boolean>,
+    ): Op<Boolean> {
+        return when (delegate) {
```

Comment:
I would use `=` instead of return here and eliminate inner returns to simplify reading